src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -267,9 +326,8 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
           RAY_LOG(FATAL) << status.ToString();
         }
       });
-  RAY_CHECK(pending_lease_requests_
-                .emplace(scheduling_key, std::make_pair(lease_client, task_id))
-                .second);
+  pending_lease_request = std::make_pair(lease_client, task_id);
+  RAY_CHECK(pending_lease_request.first);
```

Comment:
I don't think this check is necessary. The previous check was just to make sure that there wasn't already a pending lease request for the same scheduling key (arguably also not necessary).